src/table.ts

```diff
@@ -13,7 +13,7 @@ export function table(...parameters: TableParameters): string {
 
 	const trimmed = original
 		.split("\n")
-		.map((line) => line.charAt(0) + line.slice(columnWidth))
+		.map((line: string) => line.charAt(0) + line.slice(columnWidth))
```

Comment:
[Bug] This shouldn't be necessary. The `logger` import being type `any` is a problem with the types.

```suggestion
		.map((line) => line.charAt(0) + line.slice(columnWidth))
```

This gets fixed if the `.d.ts` file is changed to also named-export the `Dump` type, more like https://github.com/jimmywarting/not-a-log/blob/f28d487cfd0ac9eae7e86ca6b65bbc4db65d4d47/types.d.ts:

```diff
- type Dump = { [K in keyof Console]: ReplaceReturnType<Console[K], string> };
- var dump: Dump;
- export = dump;
+ export type Dump = { [K in keyof Console]: ReplaceReturnType<Console[K], string> };
+ export default Dump;
```

Reply:
The adding `allowJs` makes sense, I don't mean to question that. What I'm saying is that the `types.d.ts` right now isn't set up properly.

See https://github.com/JoshuaKGoldberg/console-table-without-index/commit/b60b4b86648db20b55663a8f42b02ef6f3c30ac1 for my suggested fix. That passes builds locally for me. Do you spot anything broken with it?

Reply:
Yes, it passes. I forgot to include `ReplaceReturnType` when I was testing it, my bad. So, I need to update `table.ts`, `types.d.ts` and `eslint.config.js` files, right? If so, do I have to do this in a separate commit, or can I modify an existing one?

Reply:
Go ahead and use whatever commit strategy on this branch you find the easiest 🙂. I personally normally just push new commits to the branch. But if you prefer rebasing / rewriting history, go for it.